pagopa/io-app
# feat: [IOPID-2322] - CieID Error handling

## Short description
Added error and loading handling

## List of changes proposed in this pull request
- Updated scripts `cie-ios:prod` and `cie-ios:dev`
- Replaced the default `react-native-webview` loader with the `LoadingOverlay` component in the `renderLoading` method
- Configured `onHttpError` and `onError` to use `handleLoadingError` function
- Added `isAuthenticationUrl` utility function to detect when to open the `CieID` app
- Added `handleLoginFailure` function to properly redirect to the `AuthErrorScreen`
- Added 4 dedicated errors in the `AuthErroComponent`

## Demo

<details><summary>iOS</summary>

### Operation canceled
|NFC Available|NFC not available|From Deep Link|
|--------------|-----------------|----------------|
|<video src="https://github.com/user-attachments/assets/95479823-c1f3-4533-8b9a-becbc1e59d1d" />|<video src="https://github.com/user-attachments/assets/b0e3f141-c450-40e3-b611-642a8687f629" />|<video src="https://github.com/user-attachments/assets/1326461f-4d1d-45f1-a088-6f9dfce02e6f" />|
### Invalid operation
|From Deep Link|
|----------------|
|<video src="https://github.com/user-attachments/assets/c9ab7906-6851-41bc-91e7-7b6461c503d3" />|
</details>

<details><summary>Android</summary>

### Operation canceled

|NFC Available|NFC not available|App2App|
|--------------|-----------------|--------------------|
|<video src="https://github.com/user-attachments/assets/744ad8cc-502c-4714-a5d6-41ad72aa9bd4" />|<video src="https://github.com/user-attachments/assets/39d61e8e-9e70-46b8-a8c2-97640e17ae98" />|<video src="https://github.com/user-attachments/assets/1575b45e-21ba-473c-af83-370eb8a5dbc7" />|

</details>

## How to test
Run the app in the local environment, and emulate what shown in the Demo videos

>[!Important]
>Currently, the following errors are expected:
>
>### Android
>- `CIEID_OPERATION_CANCEL`: Currently handled only on Android in the production environment.
>
>### iOS (Deep link)
>- Operation canceled
>- Operation invalid
>
>### Both platforms
>Clicking the "Cancel" CTA during the loading of the login screen.
